src/Cooked/MockChain/Testing.hs

```diff
@@ -65,180 +66,11 @@ infixr 2 .||.
-          [ "Expected two outcomes, both failed with:",
-            renderString (prettyCookedOpt pcOpts) errA,
-            "; ",
-            renderString (prettyCookedOpt pcOpts) errB,
-            "\n First: ",
-            renderString (prettyCookedOpt pcOpts) ta,
-            "\nSecond: ",
-            renderString (prettyCookedOpt pcOpts) tb
-          ]
-  xs -> testFailureMsg $ "Expected exactly two outcomes, received: " ++ show (length xs)
-
--- | Generalizes 'testBinaryRelatedBy', asserting that the given
--- 'StagedMockChain' produces more than two outcomes, say @[x,y,z,w]@, all of
--- which are successful (i.e. are not a 'MockChainError') and these states are
--- in the same equivalence class of (~); that is, they satisfy:
---
--- > x ~ y && x ~ z && x ~ z && x ~ w
---
--- Because @(~)@ should be symmetric and transitive we can estabilish that these
--- states all belong to the same equivalence class. This function does /not/
--- check each pointwise case.
-testOneEquivClass ::
-  (IsProp prop) =>
-  PrettyCookedOpts ->
-  (UtxoState -> UtxoState -> prop) ->
-  InitialDistribution ->
-  StagedMockChain a ->
-  prop
-testOneEquivClass pcOpts rel = testSatisfiesFrom' $ \case
-  [] -> testFailureMsg "Expected two of more outcomes, received: 0"
-  [_] -> testFailureMsg "Expected two of more outcomes, received: 1"
-  ((Left errX, tx) : _) -> testFailureMsg $ concat ["First outcome is a failure: ", renderString (prettyCookedOpt pcOpts) errX, "\n", renderString (prettyCookedOpt pcOpts) tx]
-  ((Right resX, _) : xs) -> go (snd resX) xs
-  where
-    -- we can flag a success here because 'xs' above is guarnateed to have at
-    -- least one element since we ruled out the empty and the singleton lists in
-    -- the \case
-    go _resX [] = testSuccess
-    go _resX ((Left errY, ty) : _) = testFailureMsg $ concat ["An outcome is a failure: ", renderString (prettyCookedOpt pcOpts) errY, "\n", renderString (prettyCookedOpt pcOpts) ty]
-    go resX ((Right (_, resY), _) : ys) = testConjoin [rel resX resY, go resX ys]
-
--- | Asserts that the results produced by running the given 'StagedMockChain'
--- from some speficied 'InitialDistribution' satisfy a given assertion. In this
--- case, the predicate gets the trace descriptions that led to each potential
--- outcome and is responsible for calling 'testCounterexample' communicate these
--- to the user.
---
--- Although this function is mainly used internally, as a building block for the
--- simpler predicates, it can be useful in building some custom
--- predicates. Check 'testAllSatisfiesFrom' or 'testBinaryRelatedBy' for
--- examples on using this.
-testSatisfiesFrom' ::
-  ([MockChainReturn a UtxoState] -> prop) ->
-  InitialDistribution ->
-  StagedMockChain a ->
-  prop
-testSatisfiesFrom' predi i0 = predi . interpretAndRunWith (runMockChainTFrom i0)
-
--- * 'TestResult' instances
-
--- Catches a HUnit test failure, if the test fails.
+-- | Catches a HUnit test failure, if the test fails.
```

Comment:
It seems we only use `assertionToMaybe` once in the instance afterwards. I don't recall ever using that as a user. Maybe this could become a `where` near its only call site.

Reply:
good point, I will perform this change.

Reply:
actually, it is used at 2 places so I can't change that